boards/arm/decawave_dwm3001c_dev/decawave_dwm3001c_dev.dts

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2023 Sahas Chitlange
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <nordic/nrf52833_qiaa.dtsi>
+#include "decawave_dwm3001c_dev-pinctrl.dtsi"
+#include <zephyr/dt-bindings/input/input-event-codes.h>
+
+/ {
+	model = "Decawave DWM3001C-DEV";
+ 	compatible = "decawave,dw3001c";
+
+	chosen {
+		zephyr,console        = &uart0;
+		zephyr,shell-uart     = &uart0;
+		zephyr,uart-mcumgr    = &uart0;
+		zephyr,bt-mon-uart    = &uart0;
+		zephyr,bt-c2h-uart    = &uart0;
+		zephyr,sram           = &sram0;
+		zephyr,flash          = &flash0;
+		zephyr,code-partition = &slot0_partition;
+		zephyr,ieee802154     = &ieee802154alias;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		led0_red: led_0 {
+			gpios = <&gpio0 14 GPIO_ACTIVE_LOW>;
+			label = "Red LED (D12)";
+		};
+		led1_red: led_1 {
+			gpios = <&gpio0 22 GPIO_ACTIVE_LOW>;
+			label = "Red LED (D11)";
+		};
+		led2_blue: led_2 {
+			gpios = <&gpio0 5 GPIO_ACTIVE_LOW>;
+			label = "Blue LED (D10)";
+		};
+		led3_green: led_3 {
+			gpios = <&gpio0 4 GPIO_ACTIVE_LOW>;
+			label = "Green LED (D9)";
+		};
+	};
+
+	buttons {
+		compatible = "gpio-keys";
+		button0: button_0 {
+			gpios = <&gpio0 2 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+			label = "Push button switch 0";
+			zephyr,code = <INPUT_KEY_0>;
+		};
+		
+	};
+
+	/* These aliases are provided for compatibility with samples */
+	aliases {
+		sw0        = &button0;
+		led0       = &led0_red;
+		led1       = &led1_red;
+		led2       = &led2_blue;
+		led3       = &led3_green;
+		led0-red   = &led0_red;
+		led1-red   = &led1_red;
+		led2-blue  = &led2_blue;
+		led3-green = &led3_green;
+		watchdog0  = &wdt0;
+		accel0     = &lis2dh12;
+	};
+
+};
+
+&adc {
+	status = "okay";
+};
+
+&uicr {
+	gpio-as-nreset;
+};
+
+&gpiote {
+	status = "okay";
+};
+
+&gpio0 {
+	status = "okay";
+};
+
+&gpio1 {
+	status = "okay";
+};
+
+&uart0 {
+	status = "okay";
+	compatible = "nordic,nrf-uart";
+	current-speed = <115200>;
+	pinctrl-0 = <&uart0_default>;
+	pinctrl-1 = <&uart0_sleep>;
+	pinctrl-names = "default", "sleep";
+};
+
+&i2c0 {
+	compatible = "nordic,nrf-twim";
+	status = "okay";
+	clock-frequency = <I2C_BITRATE_FAST>;
+
+	pinctrl-0 = <&i2c0_default>;
+	pinctrl-1 = <&i2c0_sleep>;
+	pinctrl-names = "default", "sleep";
+	lis2dh12: lis2dh12@19 {
+		compatible = "st,lis2dh12", "st,lis2dh";
+		reg = <0x19>;
+		irq-gpios = <&gpio0 16 GPIO_ACTIVE_HIGH>;
+	};
+};
+
+&spi3 {
+	compatible = "nordic,nrf-spim";
+	status = "okay";
+	cs-gpios = <&gpio1 6 GPIO_ACTIVE_LOW>;			/* P1.06 */
+
+	pinctrl-0 = <&spi3_default>;
+	pinctrl-1 = <&spi3_sleep>;
+	pinctrl-names = "default", "sleep";
+	ieee802154alias: dw3001c@0 {
```

Comment:
`devicetree error: Label 'ieee802154' appears on /soc/radio@40001000/ieee802154 and on /soc/spi@4002f000/dw3001c@0`

Kept getting that error so I changed the label name.

Reply:
The nRF52833 SoC (DWM300x) has a ieee802154 device node in its base definition while the nRF52832 (DWM100x) has not. So the ieee802154 node name is already taken (see https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/nordic/nrf52833.dtsi#L95).

Now there's [only 2 hard things in computer science](https://martinfowler.com/bliki/TwoHardThings.html) - and one of them is naming:

Boards rarely expose more than one ieee802154 capable radio which is also the reason why most drivers are not yet instance capable so we've gotten away with the ieee802154 name and single instance drivers for quite some time. But this is going to change with ranging.

In the general case, I see a few reasons why s/o would have several ieee802154 instances on a single board:
1) PHY diversity (BLE + UWB)
2) PDoA/AoA (one transceiver/driver instance per antenna if the transceiver is not multi-antenna capable - see the qorvo transceivers that cater for such a case)
3) Participating in several PANs on different channels at the same time. The standard explicitly proposes this scenario but we don't have such a requirement from the community, yet. 

[Devicetree](https://github.com/devicetree-org/devicetree-specification/releases/tag/v0.4) uses a `peripheral-name: hardware-name@memory-address` naming convention. The hardware name may be repeated (you can have several instances of the same hardware at different addresses), the peripheral name cannot as it is instance specific.

The peripheral name is usually the one from the datasheet. But not so for the ieee802154 node as it designates a sub-node of a multi-driver capable peripheral or bus.

We have one multi-driver precedence for the CC13/26xx: ieee802154 (2.4G) and ieee802154g (SubG). I don't like the one-letter postfix very much, though. It's neither scalable nor readable.

Therefore I propose the following naming scheme:
- ieee802154 for the main data communication device on a board
- ieee802154-<phy> for devices that support additional PHYs on the same board (e.g. for ranging or when translating/routing between PHYs)
- ieee802154[-<phy>]-<n> for boards with several driver instances of the same PHY for PDoA/AoA or channel diversity (n zero-based integers)

The PHY-suffix must be chosen "creatively" per board (e.g. "subg" or "sun" rather than "sun-fsk") - as it may not always cleanly map to a single PHY when a driver supports several of them. But I think we should keep it pragmatic as we'll probably always find some semantically matching string in the datasheet of a given transceiver.

@rlubos, @jukkar, @nandojve Would love to hear your opinion on those thoughts if you find the time.

Reply:
Changed the name to `ieee802154_uwb` `ieee802154-uwb` seems to throw errors, is hypen not valid?